src/OpenLoco/Windows/ScenarioSelect.cpp

```diff
@@ -0,0 +1,503 @@
+                continue;
+
+            // Highlight selected item
+            auto formatStringId = StringIds::black_stringid;
+            if (scenarioInfo == reinterpret_cast<ScenarioIndexEntry*>(self->info))
+            {
+                Gfx::drawRect(dpi, 0, y, self->width, ROW_HEIGHT - 1, 0x2000000 | 48);
+                formatStringId = StringIds::wcolour2_stringid;
+            }
+
+            // Scenario name
+            {
+                auto str = const_cast<char*>(StringManager::getString(StringIds::buffer_2039));
+                strncpy(str, scenarioInfo->scenarioName, std::size(scenarioInfo->scenarioName));
+
+                auto args = FormatArguments();
+                args.push(StringIds::buffer_2039);
+
+                const int16_t x = 210;
+                Gfx::drawStringCentred(*dpi, x, y + 1, Colour::black, formatStringId, &args);
+            }
+
+            // Completed?
+            if (!scenarioInfo->hasFlag(ScenarioIndexFlags::completed))
+            {
+                y += ROW_HEIGHT;
+                continue;
+            }
+
+            // Draw checkmark to indicate completion
+            Gfx::drawImage(dpi, 395, y + 1, 3629);
+
+            // 'Completed by' info
+            {
+                auto str = const_cast<char*>(StringManager::getString(StringIds::buffer_2039));
+                strncpy(str, scenarioInfo->highscoreName, std::size(scenarioInfo->highscoreName));
+
+                auto args = FormatArguments();
+                args.push(StringIds::completed_by_name_in_years_months);
+                args.push(StringIds::buffer_2039);
+                args.push<uint16_t>(scenarioInfo->completedMonths / 12);
+                args.push<uint16_t>(scenarioInfo->completedMonths % 12);
+
+                const int16_t x = 210;
+                Gfx::drawStringCentred(*dpi, x, y + 10, Colour::black, formatStringId, &args);
+            }
+
+            y += ROW_HEIGHT;
+        }
+    }
+
+    // 0x00443E9B
+    static void onMouseUp(window* self, const widget_index widgetIndex)
+    {
+        switch (widgetIndex)
+        {
+            case widx::close:
+                WindowManager::close(self->type);
+                break;
+        }
+    }
+
+    // 0x00443EA6
+    static void onMouseDown(window* self, widget_index widgetIndex)
+    {
+        switch (widgetIndex)
+        {
+            case widx::tab0:
+            case widx::tab1:
+            case widx::tab2:
+            case widx::tab3:
+            case widx::tab4:
+            {
+                uint8_t selectedCategory = widgetIndex - widx::tab0;
+                if (self->current_tab == selectedCategory)
+                    return;
+
+                self->current_tab = selectedCategory;
+
+                auto& config = Config::get();
+                config.scenario_selected_tab = selectedCategory;
+                Config::write();
+
+                self->object = reinterpret_cast<char*>(0xFFFFFFFF);
```

Comment:
```suggestion
                self->info = reinterpret_cast<char*>(0xFFFFFFFF);
```
Also, since you implemented the entire dialog, it maybe could also just be a nullptr?